Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -405,8 +405,8 @@ extension HTTPClient {
 
         @discardableResult
         func setChannel(_ channel: Channel) -> Channel {
-            precondition(self.eventLoop === channel.eventLoop, "Channel must use same event loop as this task.")
             return self.lock.withLock {
+                self.currentEventLoop = channel.eventLoop
```

Comment:
Oh I see you do lock it here which is good (that totally works as an alternative to the atomic I proposed) but you need to also lock it when reading it, not just when setting it here.

Reply:
Yes, I used the lock because it was already being used by the code prior to changes.

Reply:
Just fixed it using `lock` since it was previously used, but it `AtomicBox` can make things clearer I could also use it instead. Tell me what you think :)